src/cryptonote_core/service_node_list.cpp

```diff
@@ -1581,19 +1581,31 @@ namespace service_nodes
     bool result = verify_block(block, false /*alt_block*/, checkpoint);
     if (result && cryptonote::block_has_pulse_components(block))
     {
+      // NOTE: Only record participation if its a block we recently received.
+      // Otherwise processing blocks in retrospect/re-loading on restart seeds
+      // in old-data.
       uint64_t const block_height = cryptonote::get_block_height(block);
-      std::shared_ptr<const quorum> quorum = get_quorum(quorum_type::pulse, block_height, false, nullptr);
-      if (!quorum || quorum->validators.empty())
-      {
-        MERROR("Unexpected Pulse error " << (quorum ? " quorum was not generated" : " quorum was empty"));
-        return false;
-      }
+      bool newest_block           = m_blockchain.get_current_blockchain_height() == (block_height + 1);
 
-      for (size_t validator_index = 0; validator_index < service_nodes::PULSE_QUORUM_NUM_VALIDATORS; validator_index++)
+      auto now                    = pulse::clock::now().time_since_epoch();
+      auto earliest_time          = std::chrono::seconds(block.timestamp) - TARGET_BLOCK_TIME;
+      auto latest_time            = std::chrono::seconds(block.timestamp) + TARGET_BLOCK_TIME;
```

Comment:
Seems like a mild misuse of `TARGET_BLOCK_TIME` (that is: 2 minutes seems reasonable, but it seems just coincidental that we would pick the same value as the target block time).

Reply:
It does seem coincidental, but it makes some sense that you wouldn't see the next block until at least TARGET_BLOCK_TIME (and maybe some) after the top block?

Reply:
That's true, but this seems more of a comparison against the local clock: am I seeing this block within T seconds of its creation?  The top-block bit is sort of separate from that.

Reply:
Yeah the local clock bit is not exactly .. robust :p